`cargo doc` failing on stable Rust

When trying to run `cargo doc` command it fails as the `#![feature]` is not available on stable Rust versions. Here is the error message that pop ups when running the command

```
PS C:\Users\user\Documents\code\test_mavlink> cargo doc
 Documenting bitflags v2.9.1
 Documenting mavlink-core v0.15.0
error[E0554]: `#![feature]` may not be used on the stable release channel
  --> C:\Users\user\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\mavlink-core-0.15.0\src\lib.rs:23:50
   |
23 | #![cfg_attr(all(any(docsrs, doc), not(doctest)), feature(doc_auto_cfg))]
   |                                                  ^^^^^^^^^^^^^^^^^^^^^

For more information about this error, try `rustc --explain E0554`.
```

We are currently running Rust 1.87 and we would like to keep using the stable versions of Rust and not the nightly build. Is it possible to fix it as we also need the documentation tool in our project

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -128,7 +128,7 @@ jobs:
     - uses: actions/checkout@v5
     - uses: dtolnay/rust-toolchain@nightly
     - name: Build docs
-      run: cargo doc --features "default all-dialects emit-description emit-extensions format-generated-code tokio-1 signing" 
+      run: RUSTDOCFLAGS=--cfg=docsrs cargo doc --features "default all-dialects emit-description emit-extensions format-generated-code tokio-1 signing" 
     - name: Deploy
       uses: peaceiris/actions-gh-pages@v3
       if: ${{ github.ref == 'refs/heads/master' }}

--- a/mavlink-core/src/lib.rs
+++ b/mavlink-core/src/lib.rs
@@ -20,7 +20,7 @@
 //!
 //! [ardupilotmega::MavMessage]: https://docs.rs/mavlink/latest/mavlink/ardupilotmega/enum.MavMessage.html
 #![cfg_attr(not(feature = "std"), no_std)]
-#![cfg_attr(all(any(docsrs, doc), not(doctest)), feature(doc_auto_cfg))]
+#![cfg_attr(docsrs, feature(doc_auto_cfg))]
 #![deny(clippy::all)]
 #![warn(clippy::use_self)]
 

--- a/mavlink/src/lib.rs
+++ b/mavlink/src/lib.rs
@@ -1,5 +1,5 @@
 #![cfg_attr(not(feature = "std"), no_std)]
-#![cfg_attr(all(any(docsrs, doc), not(doctest)), feature(doc_auto_cfg))]
+#![cfg_attr(docsrs, feature(doc_auto_cfg))]
 
 //! Rust implementation of the MAVLink UAV messaging protocol, with bindings for all dialects.
 //! This crate provides message set code generation, packet building, parsing and connection handling for blocking and asynchronous I/O.

